installer/install.go

```diff
@@ -0,0 +1,397 @@
+//	if opts.ImageRepoUser == "" {
+//		var err error
+//		opts.ImageRepoUser, err = getDockerUsername()
+//		if err != nil {
+//			return err
+//		}
+//	}
+//
+//	if opts.ImageRepoPw == "" {
+//		var err error
+//		opts.ImageRepoPw, err = getDockerPw()
+//		if err != nil {
+//			return err
+//		}
+//	}
+//
+//	var authConfig = dockerTypes.AuthConfig{
+//		Username: opts.ImageRepoUser,
+//		Password: opts.ImageRepoPw,
+//	}
+//	authConfigBytes, err := json.Marshal(authConfig)
+//	if err != nil {
+//		return err
+//	}
+//	authConfigEncoded := base64.URLEncoding.EncodeToString(authConfigBytes)
+//	pushOpts := dockerTypes.ImagePushOptions{RegistryAuth: authConfigEncoded, All: true}
+//
+//	for _, image := range images {
+//		log.Info("Pushing image")
+//
+//		ctx, cancel := context.WithTimeout(context.Background(), time.Minute*10)
+//		defer cancel()
+//
+//		reader, err := dockerClient.ImagePush(ctx, image, pushOpts)
+//		defer reader.Close()
+//		if err != nil {
+//			return err
+//		}
+//
+//		scanner := bufio.NewScanner(reader)
+//		for scanner.Scan() {
+//			response := DockerPushResponse{}
+//			err := json.Unmarshal(scanner.Bytes(), &response)
+//			if err != nil {
+//				return err
+//			}
+//
+//			if response.Error != "" {
+//				log.WithFields(log.Fields{
+//					"error":       response.Error,
+//					"errorDetail": *response.ErrorDetail,
+//				}).Error("Docker load error")
+//				return fmt.Errorf("error loading images")
+//			}
+//
+//			log.WithFields(log.Fields{
+//				"status": response.Status,
+//				"id":     response.Id,
+//			}).Info("Pushing image...")
+//		}
+//	}
+//	return nil
+//}
+
+func createNamespace(ns string) (string, error) {
+	cmd := exec.Command("kubectl", "create", "ns", ns)
+	return runCmd(cmd)
+}
+
+func runCmd(cmd *exec.Cmd) (string, error) {
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return string(output), fmt.Errorf("%s: %s", err, output)
+	}
+	return string(output), nil
+}
+
+type ErrorDetail struct {
+	Message string `json:"message"`
+}
+
+type DockerPushResponse struct {
+	Status      string       `json:"status"`
+	Id          string       `json:"id"`
+	Error       string       `json:"error"`
+	ErrorDetail *ErrorDetail `json:"errorDetail"`
+}
+
+type DockerLoadResponse struct {
+	Stream      string       `json:"stream"`
+	Error       string       `json:"error"`
+	ErrorDetail *ErrorDetail `json:"errorDetail"`
+}
+
+type Options struct {
+	ImageRepo         string `short:"r" long:"image-repo" required:"false" description:"Private Docker image repo URL" value-name:"URL"`
+	ImageTar          string `short:"p" long:"image-tar" required:"false" description:"Path to image tar" value-name:"PATH"`
+	//ImageRepoUser     string `long:"repo-user" required:"false" description:"Private Docker image repo URL" value-name:"USER"`
+	//ImageRepoPw       string `long:"repo-pw" required:"false" description:"Private Docker image repo URL" value-name:"PASSWORD"`
```

Comment:
These will come back if I can get the go-docker package working with image push. 
It is making a system call to push, currently.